src/components/progress/Progress.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+    <div class="progress-wrapper">
+        <progress
+            ref="progress"
+            class="progress"
+            :class="newType"
+            :max="max">{{ newValue }}</progress>
+        <p
+            v-if="showValue"
+            class="progress-value has-text-white"><slot>{{ newValue }}</slot></p>
+    </div>
+</template>
+
+<script>
+
+export default {
+    name: 'BProgress',
+    props: {
+        type: {
+            type: String | Object,
+            default: 'is-darkgrey'
+        },
+        size: String,
+        value: {
+            type: undefined | Number,
+            default: undefined
+        },
+        max: {
+            type: Number,
+            default: 100
+        },
+        showValue: {
+            type: Boolean,
+            default: false
+        },
+        format: {
+            type: String,
+            default: 'raw',
+            validator: (value) => {
+                return [
+                    'raw',
+                    'percent'
+                ].indexOf(value) >= 0
+            }
+        },
+        precision: {
+            type: Number,
+            default: 2
+        },
+        keepTrailingZeroes: {
+            type: Boolean,
+            default: false
+        }
+    },
+    computed: {
+        isIndeterminate() {
+            return this.value === undefined || this.value === null
+        },
+        newType() {
+            return [
+                this.size,
+                this.type
+            ]
+        },
+        newValue() {
+            if (this.value === undefined || this.value === null || isNaN(this.value)) {
+                return undefined
+            }
+
+            if (this.format === 'percent') {
+                const val = this.toFixed(this.value * this.max / 100)
+                return `${val}%`
+            }
+            const val = this.toFixed(this.value)
+            return val
+        }
+    },
+    watch: {
+        value(value) {
+            this.setValue(value)
```

Comment:
In order to remove the `setValue` call in mounted hook, you might add `immediate` option in the watcher, for example:
```
value: {
    immediate: true,
    handler(value) {
      ...
    }
  }
```

Reply:
The problem by using the `immediate` property is the function will be called before vue's mount hook. That means that $refs won't be visible from the watch function.
You can see my next comment about the usage of $refs.

Reply:
You're right!